### ISSUE TITLE
[Enhancement]: Add cart line items / total information to the StoreApi checkout endpoint

### ISSUE THREAD

**Issue by phil-sola** [woothemes/woocommerce#53279]
### Describe the solution you'd like

The data returned from the **wc/store/v1/checkout** endpoint can be seen [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/StoreApi/docs/checkout.md).

It's incredibly basic and not entirely helpful.

Typically on a Checkout page you would have the order information displayed with line items, tax information, shipping, total cost etc. Something like this as an example (taken from the Woocommerce official docs):

![Image](https://github.com/user-attachments/assets/2188562f-f22e-45de-881b-482127e34d8f)

Currently if sending a REST request to the above checkout endpoint I get the billing address, shipping address, order_id and order_key for fetching the draft order created, but that means sending off a 2nd pointless REST req. when that information is always needed at checkout.

If users go from cart -> checkout in a linear fashion that's fine I guess, I could send the data along with the page req, but users may shop around more and then go straight to checkout resulting in multiple requests needing to be made.

I would like to see the cart data sent back from the checkout GET request. This makes the endpoint far more useful for 99.9% of cases and will result in less API requests to the Woo site.

### Describe alternatives you've considered

I could use the **wc/store/v1/cart**  (seen [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/StoreApi/docs/cart.md)) endpoint but that defeats the purpose of having a seperate endpoint for checkout.

Either the checkout GET endpoint is needed or not. If it is, then it should contain the same helpful information the cart GET endpoint does to save multiple requests.

**Comment by senadir** [comment]
> endpoint but that defeats the purpose of having a seperate endpoint for checkout.

I disagree on this point, the Checkout endpoint is responsible for user information, checkout related data (like soon managing order only fields), and passing back payment data.

I think ideally we should allow batching both requests on the same request. We might be looking into adding more data to Checkout endpoint (or possibly including cart in it) but that's an open question that I believe we explored and couldn't reach a conclusion.

**Comment by phil-sola** [comment]
Hey @senadir, appreciate the comment and happy to disagree on the mentioned comment, but 100% do believe that either batching these requests or adding more data is needed.

What should be super simple and ideally a single req. isn't at the moment and that definitely needs exploring.

I hope you can all reach a conclusion on it fairly soon - I don't really see any negative downsides to adding this data to the checkout endpoint.

**Comment by senadir** [comment]
Hey @phil-sola, just so you know, we're going to introduce a `PUT /checkout` endpoint that also includes Cart, if it works well, we will extend cart to the GET request as well

https://github.com/woocommerce/woocommerce/pull/53553

**Comment by phil-sola** [comment]
Hey @senadir sorry for the delay coming back to you, I've been sick for the last week.

Thanks so much for the update and keeping me posted, that's really exciting!

**Comment by jamesckemp** [comment]
Hi @phil-sola - thanks for reporting this, and for the detail you gave.

I know this has been open a while, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

It looks like this has already been fixed, or it's no longer happening on trunk.

Here's what I found: Current trunk checkout responses include __experimentalCart through CheckoutSchema::get_checkout_response() when a cart is passed and through get_draft_response() for no-order checkout responses. GET /wc/store/v1/checkout uses build_draft_route_response() when no draft order exists, so it now returns cart data from the current cart path.

I'm going to close this one. If you're still facing this issue on a current version, please reply with steps to reproduce and we'll happily take another look.

Fix/changelog reference: https://github.com/woocommerce/woocommerce/pull/53553
